Sprint_6

iasem84 opened the pull request.

iasem84 committed:
Sprint_6 first version

okryukova commented on src/main/java/com/example/Lion.java:
```diff
@@ -2,31 +2,33 @@
 import java.util.List;
 
-public class Lion {
+public class Lion extends Feline{
```
⛔️Нужно исправить.  Наследование здесь не нужно

okryukova commented on src/main/java/com/example/Lion.java:
```diff
@@ -2,31 +2,33 @@
+public class Lion extends Feline{
 
     boolean hasMane;
+    Predator predator;
```
⛔️Нужно исправить.  Принцип инъекции зависимостей реализован неверно. Объект должен быть класса Feline

okryukova commented on src/test/java/CatMockTest.java:
```diff
@@ -0,0 +1,29 @@
+    Feline feline;
+
+    @Test
+    public void exceptionTextCorrectText() throws Exception {
```
⛔️Нужно исправить. В названии метода необходимо использовать слово Test. Проверим, соблюдается ли данное правило для всех созданных тестов.

okryukova replied:
⚠️Можно улучшить. Название теста неинформативно. О каком тексте идет  речь? Поправь, пожалуйста, и названия других тестов

okryukova commented on src/test/java/CatMockTest.java:
```diff
@@ -0,0 +1,29 @@
+        Mockito.when(feline.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
+        List<String> expectedResult = List.of("Животные", "Птицы", "Рыба");
+        List<String> actualResult = cat.getFood();
+        assertEquals(expectedResult, actualResult);
```
⚠️Можно улучшить. Можно использовать поясняющие сообщения для assert. Тогда будет проще разобраться, почему тест упал.

okryukova commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,34 @@
+
+import static org.junit.Assert.assertEquals;
+
+public class CatTest {
```
⛔️Нужно исправить. Тесты для класса Cat нужно объединить

okryukova commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,34 @@
+
+public class CatTest {
+
+    Feline feline;
```
⛔️Нужно исправить. Нужно использовать mock , а не реальный объект

okryukova commented on src/test/java/FelineMockTest.java:
```diff
@@ -0,0 +1,19 @@
+public class FelineMockTest {
+
+    @Spy
+    Feline feline;
```
⛔️Нужно исправить.  Создавай spy только для зависимостей, а не для тестируемого объекта.

okryukova commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,38 @@
+
+import static org.junit.Assert.assertEquals;
+
+public class FelineTest {
```
⛔️Нужно исправить. Тесты для класса Feline нужно объединить

okryukova commented on src/test/java/LionParamTest.java:
```diff
@@ -0,0 +1,38 @@
+@RunWith(Parameterized.class)
+public class LionParamTest {
+    Lion lion;
+    Feline feline;
```
⛔️Нужно исправить.  Для зависимостей нужно использовать моки, а не реальные объекты. Поправь, пожалуйста, этот момент во всем коде

okryukova commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,49 @@
+    @Test
+    public void exceptionTextTest() {
+        String expectedMessage = "Используйте допустимые значения пола животного - самец или самка";
+        try {
```
⛔️Нужно исправить. В Junit4 есть метод assertThrows и @test(expected = Exception.class) для проверки исключений, try-cath нелучшая практика в тестах.

iasem84 committed:
1. Убрано наследование в классе Lion и исправлена реализация инъекции зависимости.
2. Исправление опечатка в названии метода exceptionTextCorrectText(), а также переименован сам метод
3. Добавлены поясняющие сообщения для assert.
4. Объединены тесты для классов Cat и Feline
5. Использованы моки для зависимостей
6. Использован метод assertThrows для проверки текста исключения

okryukova reviewed (approved):
Отличная работа!
